### ISSUE TITLE
Replace `&nbsp;` with spaces on the note editor

### ISSUE THREAD

**Issue by BrayanDSO** [ankidroid/Anki-Android#12685]
**Is your feature request related to a problem? Please describe.**
Sometimes when I edit a note on Anki desktop then sync, some spaces will appear as `&nbsp;`, which is quite suffarable to see and edit

**Describe the solution you'd like**
Add an option to replace `&nbsp;` with spaces visually, similar to the option to replace `<br>` with newlines visually

**Describe alternatives you've considered**
Keep deciphering HTML escaped spaces

**Additional context**

**Comment by mikehardy** [comment]
believe it or not this is a frequent user complaint

**Comment by oakkitten** [comment]
I remember this being discussed on the forums, and @kleinerpirat made an addon that fixes it https://forums.ankiweb.net/t/nbspremover-official-support/20327/11

**Comment by criticalAY** [comment]
`s = s.replace("<br>", " ")` should we do something similar to this ?

**Comment by oakkitten** [comment]
Non-breaking spaces have legitimate uses (which are the reason for their very existence). A proper solution would be to make Anki not to insert them, but user decks have accumulated a lot of those so since we show HTML maybe having some sort of a way to not show these would be reasonable.

I suppose that if you simply just replace them with spaces or U+00A0, you'd produce a lot of unwanted edits since neither of those equal `&nbsp;`. I think a better way would be make a special `Span` that is `&nbsp;` under the hood but is displayed as a space, or perhaps some sort of a symbol such as `␣`.

**Comment by BrayanDSO** [comment]
Closing this in favor of spending time in a WYSIWYG editor
